doc/sphinx/language/core/records.rst

```diff
@@ -3,208 +3,313 @@
-  Eval compute in bottom half.
-  Eval compute in Rat_bottom_cond half.
+   .. insertprodn term_projection term_projection
 
-An alternative syntax for projections based on a dot notation is
-available:
+   .. prodn::
+      term_projection ::= @term0 .( @qualid {? @univ_annot } {* @arg } )
+      | @term0 .( @ @qualid {? @univ_annot } {* @term1 } )
 
-.. coqtop:: all
+   The value of a field can be accessed using *projection form* (:n:`@term_projection`,
+   shown here) or with *application form* (see :n:`@term_application`) using the
+   projection function associated with the field.  Don't forget the parentheses for the
+   projection form.
+   Glossing over some syntactic details, the two forms are:
 
-   Eval compute in half.(top).
+   - :n:`@qualid__record.( {? @ } @qualid__field {* @arg })` (projection) and
 
-.. flag:: Printing Projections
+   - :n:`{? @ } @qualid__field {* @arg } @qualid__record` (application)
 
-   This :term:`flag` activates the dot notation for printing.
+   where the :n:`@arg`\s are the parameters of the inductive type.  If :n:`@` is
+   specified, all implicit arguments must be provided.
 
-   .. example::
+   Since the projected object is part of the notation, it is always
+   considered an explicit argument of :token:`qualid`, even if it is
+   formally declared as implicit (see :ref:`ImplicitArguments`).
+
+   .. exn:: @ident cannot be defined because it is informative and @ident is not
+
+      For example, :n:`Record foo:Prop := { x:Type }` generates the message
+      "x cannot be defined ... and foo is not".  Proofs (objects of sort :n:`Prop`)
+      are supposed to be non-distinguishable.  If you have two inhabitants of
+      :n:`Type`, such as :n:`%{%| x := nat %|%}` and :n:`%{%| x := bool %|%}`, they are
+      distinguishable (i.e. informative) and are therefore prohibited.
+
+   .. example:: Accessing record fields
 
       .. coqtop:: all
 
-         Set Printing Projections.
-         Check top half.
+         (* projection form *)
+         Eval compute in half.(top).
+
+      .. coqtop:: in
+
+         Goal True.
+
+      .. coqtop:: all
+
+         let x := eval compute in half.(top) in idtac x.
+
+         (* application form *)
+         Eval compute in top half.
+         Eval compute in bottom half.
+         Eval compute in Rat_bottom_nonzero half.
+
+   .. example:: Matching on records
+
+      .. coqtop:: all
+
+         Eval compute in (
+           match half with
+           | {| negative := false; top := n |} => n
+           | _ => 0
+           end).
+
+   .. example:: Accessing anonymous record fields with match
+
+      .. coqtop:: in
+
+         Record T := const { _ : nat }.
+         Definition gett x := match x with const n => n end.
+         Definition inst := const 3.
+
+      .. coqtop:: all
+
+         Eval compute in gett inst.
+
 
-.. FIXME: move this to the main grammar in the spec chapter
+The following settings let you control the display format for record types:
```

Comment:
Apart from the `Printing Projection` flag, it seems to me that these flags and tables should be closer to the part on defining record types (and the note that was added there, maybe even moved to be part of the note).

Reply:
I moved the notes and warnings into the record definition section.  The flags are all related to printing, so it seems reasonable to have them after the "Accessing" section.  Since it's short, I lean against making it a separate subsection, though I'll do that if you prefer.

Reply:
I'm not so fond of having things at the end of a section that do not semantically belong to the last subsection...